proposals/0000-splice-imports.rst

```diff
@@ -33,6 +33,24 @@ actually be used in a top-level splice.
    for all dependencies.
 4. When cross-compiling, all packages need to be compiled for both host and target.
 
+Definitions
+-----------
+
+level
+  Each expression exists at a level. The level is increased by 1 when
+  inside a quote and decreased by 1 inside a splice. Therefore the level of
+  an expression can be calcated as the number of quotes surrounding an expression
+  subtract the number of splices.
+
+top-level splice
+  A splice, where the body is at a negative level or a top-level unadorned
```

Comment:
I'm confused by "or" here. It seems that the first set includes the second, making the "or" part redundant. Or am I missing something?

Reply:
I don't know how to describe a declaration splice which is not explicitly spliced. For example

```
module M where

import Lens

data A ...

deriveLens 'A
```

The call to `deriveLens 'A` should be understood to be at a negative level (hence `deriveLens` should be `splice` imported) but it isn't surrounded by splices.